libutils/set.c

```diff
@@ -200,3 +200,48 @@ JsonElement *StringSetToJson(const StringSet *set)
 
     return arr;
 }
+
+static bool VisitJsonArrayFirst(ARG_UNUSED JsonElement *element, void *data)
+{
+    /* We only want one array with items, so just make sure there are no items
+     * in the string set yet. This doesn't fail if there is a nested array as
+     * the first item in the top-level array, but let's just live with that for
+     * the sake of simplicity. */
+    StringSet *set = data;
+    return (StringSetSize(set) == 0);
+}
+
+static bool AddArrayItemToStringSet(JsonElement *element, void *data)
+{
+    char *element_str = JsonPrimitiveToString(element);
+    StringSet *set = data;
+    if ((element_str != NULL) && (set != NULL))
+    {
+        StringSetAdd(set, element_str);
+        return true;
+    }
+    return false;
```

Comment:
@vpodzime This function doesn't handle `data == NULL` correctly - it will leak the `element_str`. Since you have a comparison for `(set != NULL)` it looks like it is supposed to handle it. 

It should start with one of these 2:

```
assert(data != NULL);
```

```
if (data == NULL)
{
    return false;
}
```

Not a big issue of course, it is currently always called with non-null data.